app/models/media_attachment.rb

```diff
@@ -336,8 +336,8 @@ def check_video_dimensions
 
     return unless movie.valid?
 
-    raise Mastodon::DimensionsValidationError, "#{movie.width}x#{movie.height} videos are not supported" if movie.width * movie.height > MAX_VIDEO_MATRIX_LIMIT
-    raise Mastodon::DimensionsValidationError, "#{movie.frame_rate.to_i}fps videos are not supported" if movie.frame_rate > MAX_VIDEO_FRAME_RATE
+    raise Mastodon::DimensionsValidationError, "#{movie.width}x#{movie.height} videos are not supported" if movie.width.nil? || movie.width * movie.height > MAX_VIDEO_MATRIX_LIMIT
```

Comment:
Hm, this will raise error "x videos are not supported". I am not sure if it is necessary to raise these errors if there is no video stream. Perhaps raise another error like "Video has no video stream" before if `movie.width.nil? || movie.frame_rate.nil?`?

Reply:
That sounds better. Change.